scalableminds/webknossos
# Apply d/f switching option to copying slices

This PR changes the copy slice volume functionality so that the considered "previous" slice will always be z-1 when the d/f option is disabled.
Additionally, this PR adds a small fix that keeps the `spaceDirectionOrtho` value of the direction, where it did not change. Previously, if a slice was annotated twice, the second update always caused the dimension that did not change (e.g. z coordinate) to be 1. Now the value the previous value will be kept. -> (This is not so easy to test without using console output)

### URL of deployed dev instance (used for testing):
- https://fixdirectionwhencopyingvolume.webknossos.xyz

### Steps to test:
- Open a volume tracing without an existing volume layer.
- Just use the x-y viewport for this and turn off d/f switching:
1. annotate something
2. move to the next slice and use `v` to copy the annotation
3. Then erase the copied annotation and move back to the previously annotated slice
4. Annotate this slice a bit further and the move to the next slice again
5. copy that slice using `v`. The slice should be copied.
6. Move back to the first slice, erase everything, and use `shift + v` to copy the "second" slice. This should work.


### Issues:
- fixes  #4708

------
